pkg/server/sandbox_stop.go

```diff
@@ -63,22 +63,27 @@ func (c *criContainerdService) StopPodSandbox(ctx context.Context, r *runtime.St
 	}
 
 	// Teardown network for sandbox.
-	_, err = c.os.Stat(sandbox.NetNS)
-	if err == nil {
-		if !sandbox.Config.GetLinux().GetSecurityContext().GetNamespaceOptions().GetHostNetwork() {
-			if teardownErr := c.netPlugin.TearDownPod(ocicni.PodNetwork{
-				Name:         sandbox.Config.GetMetadata().GetName(),
-				Namespace:    sandbox.Config.GetMetadata().GetNamespace(),
-				ID:           id,
-				NetNS:        sandbox.NetNS,
-				PortMappings: toCNIPortMappings(sandbox.Config.GetPortMappings()),
-			}); teardownErr != nil {
-				return nil, fmt.Errorf("failed to destroy network for sandbox %q: %v", id, teardownErr)
-			}
+	if sandbox.NetNSPath != "" {
+		if _, err := os.Stat(sandbox.NetNSPath); err != nil {
+			return nil, fmt.Errorf("failed to stat network namespace path %s :%v", sandbox.NetNSPath, err)
 		}
-	} else if !os.IsNotExist(err) { // It's ok for sandbox.NetNS to *not* exist
-		return nil, fmt.Errorf("failed to stat netns path for sandbox %q before tearing down the network: %v", id, err)
+		if teardownErr := c.netPlugin.TearDownPod(ocicni.PodNetwork{
+			Name:         sandbox.Config.GetMetadata().GetName(),
+			Namespace:    sandbox.Config.GetMetadata().GetNamespace(),
+			ID:           id,
+			NetNS:        sandbox.NetNSPath,
+			PortMappings: toCNIPortMappings(sandbox.Config.GetPortMappings()),
+		}); teardownErr != nil {
+			return nil, fmt.Errorf("failed to destroy network for sandbox %q: %v", id, teardownErr)
+		}
```

Comment:
Mentioned above https://github.com/kubernetes-incubator/cri-containerd/pull/138#discussion_r134025419.
Please add the TODO.

`TearDownPod` is not idempotent. If we successfully `TearDownPod`, but fail to remove network namespace, we'll never be able to stop the sandbox, because `TearDownPod` will always return error.

This is a bug which did happen and caused trouble to us, we should fix it soon. Let's add the TODO here.